### ISSUE TITLE
 quantile prediction probabilities in the TFT architecture

### ISSUE THREAD

**Issue by smorf-ntsg** [jdb78/pytorch-forecasting#556]
When using TFT to extract quantiles during prediction, what probabilities are returned? 

In the documentation, it appears the following breaks are used for the quantileLoss:

> quantiles:` List[float] = [0.02, 0.1, 0.25, 0.5, 0.75, 0.9, 0.98],

Do these values correspond to the output quantiles, assuming output_size = 7?

Is there a way to query the model to see which probabilities are being reported in the case I were to specify output_size as either 3 or 5, or other non-default value? 
 
Thanks, I hope I didn't miss this point in the documentation. Pytorch-forecasting is a great tool!

**Comment by jdb78** [comment]
>Do these values correspond to the output quantiles, assuming output_size = 7?

Yes

> Is there a way to query the model to see which probabilities are being reported in the case I were to specify output_size as either 3 or 5, or other non-default value?

Just use model.loss.quantiles. You can also pass a different `QuantileLoss(quantiles=[...])` to regress different quantiles.

**Comment by mmcux** [comment]
I wanted to map the predictions to their quantile and was quite surprised that for some predictions i.e. the 0.02 quantile value was higher than the 0.1 quantile value.

| Tables        | Are           | Cool  |
---
This pattern occured on my own data as well as in the example get_stallion_data.

How to reproduce

**Comment by mmcux** [comment]
I wanted to map the predictions to their quantile and was quite surprised that for some predictions i.e. the 0.02 quantile value was higher than the 0.1 quantile value.

---
This pattern occured on my own data as well as in the example get_stallion_data.

How to reproduce on the example dataset with the TemporalFusionTransformer: I copied the example until:
```
tft = TemporalFusionTransformer.from_dataset(
    training,
    learning_rate=0.03,
    hidden_size=16,
    attention_head_size=1,
    dropout=0.1,
    hidden_continuous_size=8,
    output_size=7,  # 7 quantiles by default
    loss=QuantileLoss(),
    log_interval=10,  # uncomment for learning rate finder and otherwise, e.g. to 10 for logging every 10 batches
    reduce_on_plateau_patience=4,
)
```
```
trainer.fit(
    tft,
    train_dataloader=train_dataloader,
    val_dataloaders=val_dataloader,
)
```

`tft.loss.quantiles` gives [0.02, 0.1, 0.25, 0.5, 0.75, 0.9, 0.98] as expected

A simple query like
```
quant_predictions = tft.predict(val_dataloader, mode="quantiles")
sorted_wrong = 0
for prediction in quant_predictions:
    for timestep in prediction:
        if not torch.all(timestep[:-1] <= timestep[1:]):
            print(timestep)
            number_sorted_wrong += 1
sorted_wrong
```
shows that in a couple of predictions this pattern occurs. These rows look like this: `tensor([ 554.5942,  545.1486,  811.6405, 1042.6752, 1112.8491, 1519.5549,
        2112.5273]), tensor([ 6.2597,  6.0438, 10.5865, 14.2585, 16.2907, 21.9303, 32.0497])`

I can't reproduce this always although the pytorch lightning seed is set (as in the example).


- ubuntu 20.04
- pytorch-forecasting=0.9.0

Is my interpretation of the predictions wrong or could it be a bug?

**Comment by sidd-realtor** [comment]
Hey @mmcux ,
Did you find the reason for this? I am facing the same problem.

**Comment by jecjecj** [comment]
Any updated on this? Facing the same problem here.
